File: solutions/julia/collatz-conjecture/1/collatz-conjecture.jl
function collatz_steps(n)
    if n < 1
        throw(DomainError(n))
    end

    steps = 0
    while n > 1
        if n % 2 == 0
            n = n / 2
        else
            n = 3n + 1
        end
        steps += 1
    end
    steps
end

File: solutions/clojure/collatz-conjecture/1/src/collatz_conjecture.clj
(ns collatz-conjecture)

(defn collatz-rule [n]
  (if (even? n)
      (/ n 2)
    (inc (* n 3))))

(defn collatz
  "Returns the number of steps for num to reach 1
  according to the Collatz Conjecture."
  [num]
  (loop [steps 0 n num]
    (if (= n 1)
        steps
      (recur (inc steps) (collatz-rule n)))))

